akka-remote/src/main/scala/akka/remote/artery/RemotingFlightRecorder.scala

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright (C) 2018-2019 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.remote.artery
+
+import java.net.InetSocketAddress
+
+import akka.actor.Address
+import akka.actor.ExtendedActorSystem
+import akka.actor.Extension
+import akka.actor.ExtensionId
+import akka.actor.ExtensionIdProvider
+import akka.annotation.InternalApi
+import akka.remote.UniqueAddress
+import akka.util.JavaVersion
+
+import scala.util.Failure
+import scala.util.Success
+
+/**
+ * INTERNAL API
+ */
+@InternalApi
+object RemotingFlightRecorder extends ExtensionId[RemotingFlightRecorder] with ExtensionIdProvider {
+
+  override def createExtension(system: ExtendedActorSystem): RemotingFlightRecorder =
+    if (JavaVersion.majorVersion >= 11) {
+      // Dynamic instantiation to not trigger class load on earlier JDKs
+      system.dynamicAccess.createInstanceFor[RemotingFlightRecorder](
+        "akka.remote.artery.jfr.JFRRemotingFlightRecorder",
+        (classOf[ExtendedActorSystem], system) :: Nil) match {
+        case Success(jfr) => jfr
+        case Failure(_) =>
+          system.log.debug("Failed to load JFR remoting flight recorder, falling back to noop")
```

Comment:
would be good to include the exception in this logging, and maybe also log it at warning level since it's not supposed to fail